kedro-datasets/kedro_datasets/pandas/excel_dataset.py

```diff
@@ -257,3 +257,11 @@ def _invalidate_cache(self) -> None:
         """Invalidate underlying filesystem caches."""
         filepath = get_filepath_str(self._filepath, self._protocol)
         self._fs.invalidate_cache(filepath)
+
+    def _preview(self, nrows: int = 40) -> Dict:
+        # Create a copy so it doesn't contaminate the original dataset
+        dataset_copy = self._copy()
+        dataset_copy._load_args["nrows"] = nrows  # pylint: disable=protected-access
+        data = dataset_copy.load()
+
+        return data.to_dict(orient="split")
```

Comment:
I don't know do if we already have a test covering this. Will be good to have (in framework) but not a must for this PR.

Something like this

```python
def test_copy_not_mutate_original_dataset(raw_dataset):
  copy = raw_dataset._copy()
  copy.load()
  raw_dataset == deepcopy(raw_dataset)  # This won't work yet but the idea is to check the copy doesn't interfere the raw_dataset
```

Reply:
thanks for the suggestion, if we can do it separately it would be great. Shall I create a ticket/issue for this?